bson/primitive/decimal.go

```diff
@@ -52,13 +59,13 @@ func (d Decimal128) String() string {
 	if d.h>>61&3 == 3 {
```

Comment:
I was unable to put this comment on the lines I am actually referring to (L43-45), but here goes:

This is outside of the scope of your patch, but one of the major issues with this code was that the variable names was non-descriptive one-letter variables, which is precisely why those doc comments have to be placed next to the variables `pos`, `e`, `h` and `l`.

A simple `gorename` may be what we want here to make it so that we rename them to more clear names, e.g., `e` -> `exp` (noted below), `pos` -> `posSign`, `h` -> `highSignificand`, etc.